common/src/backends/tracing.rs

```diff
@@ -67,15 +74,47 @@ pub struct ExtractPropagation<S> {
     inner: S,
 }
 
+#[pin_project]
+pub struct ExtractPropagationFuture<F> {
+    #[pin]
+    response_future: F,
+    span: Span,
+}
+
+impl<F, Body, Error> Future for ExtractPropagationFuture<F>
+where
+    F: Future<Output = Result<Response<Body>, Error>>,
+{
+    type Output = Result<Response<Body>, Error>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.project();
+
+        let _guard = this.span.enter();
```

Comment:
I'm wondering if it's possible to enter a span multiple times, or if it will crash here?

Reply:
My thinking is that since this is not an `async` block, when this scope is exited the EnterGuard will get dropped and the span exited. And then it will enter again when it's polled again. I'm not a 100% on this though, I'll try to find a definitive answer.

Reply:
Went for instrumenting the future instead, so it will be entered each time it's polled, without needing to pass the span in the future struct.